### ISSUE TITLE
Cleanup ?

### ISSUE THREAD

**Issue by kyrios123** [projectM-visualizer/projectm#95]
I have just built the version 3.0.0 and when I look at what has been installed I see things like this:
```
/usr/share/projectM/presets/!!!---flexi + amandio c - organic12-3d-2.milk
/usr/share/projectM/presets/.DS_Store
/usr/share/projectM/presets/._.DS_Store
/usr/share/projectM/presets/._Aderrasi + Geiss - Airhandler (Kali Mix) - Canvas Mix.milk
/usr/share/projectM/presets/._Aderrasi - Accelerator (Hot Lead Transfusion).milk
/usr/share/projectM/presets/._Aderrasi - Arcworld (Arcwheel - Train Wheel).milk
/usr/share/projectM/presets/._Aderrasi - Calabi-Jau Space Bar.milk
/usr/share/projectM/presets/._Aderrasi - Contortion (Flair Mix).milk
/usr/share/projectM/presets/._Aderrasi - Potion of Spirits.milk
/usr/share/projectM/presets/._Eo.S.+Phat Fractical_dancer - light in the distance.milk
/usr/share/projectM/presets/._Eo.S.+Phat Quadrent_fractal.milk
/usr/share/projectM/presets/._Eo.S.+Phat Speak with the orb_more_colour_mix.milk
/usr/share/projectM/presets/._Eo.S.+Phat last of it's kind_sinking.milk
/usr/share/projectM/presets/._Flexi - reality tunnel.milk
/usr/share/projectM/presets/._Flexi - smashing fractals [acid etching mix].milk
/usr/share/projectM/presets/._Fvese - Round and Round (geiss gamma mix).milk
/usr/share/projectM/presets/._Fvese - The Tunnel (Final Stage Mix).milk
/usr/share/projectM/presets/._Geiss - Aieeeeee!!!.milk
/usr/share/projectM/presets/._Geiss - Asymptote.milk
/usr/share/projectM/presets/._Geiss - Bass Kaleidosphere.milk
/usr/share/projectM/presets/._Geiss - Bass Zoom.milk
/usr/share/projectM/presets/._Geiss - Bipolar 1.milk
/usr/share/projectM/presets/._Geiss - Bonfire.milk
/usr/share/projectM/presets/._Geiss - Calligraphy.milk
/usr/share/projectM/presets/._Geiss - Cartographie.milk
/usr/share/projectM/presets/._Geiss - Cepiasound.milk
/usr/share/projectM/presets/._Geiss - Constant Velocity.milk
/usr/share/projectM/presets/._Geiss - Iris.milk
/usr/share/projectM/presets/._Geiss - Music Box.milk
/usr/share/projectM/presets/._Image415.jpg
/usr/share/projectM/presets/._Phat and Eo.S. _ shapes are cool smoke move2.milk
/usr/share/projectM/presets/._Phat_Eo.S._Swim_waveform_mix.milk
/usr/share/projectM/presets/._Phat_Zylot_Eo.S. spiral_faces_ravers_phantom.milk
/usr/share/projectM/presets/._PieturP - triptrap_(ultimate-trip-mix).milk
/usr/share/projectM/presets/._Zylot + Geiss - Enlightenment.milk
/usr/share/projectM/presets/._Zylot - Age of Science (seeking truth mix).milk
/usr/share/projectM/presets/._Zylot - Block Of Sound (Abstract Architecture Mix).milk
/usr/share/projectM/presets/._Zylot - Color Of Music.milk
/usr/share/projectM/presets/._Zylot - In death there is life (Dancing Lights mix).milk
/usr/share/projectM/presets/._Zylot - In death there is life (Geiss Layered Mix).milk
/usr/share/projectM/presets/._fiShbRaiN - white scream firefly.milk
/usr/share/projectM/presets/._fiShbRaiN - witchcraft.milk
/usr/share/projectM/presets/._kaite.jpg
/usr/share/projectM/presets/._manyfish.jpg
/usr/share/projectM/presets/._onefish.jpg
```
Are these files required or is that some rubbishes that can be deleted ?

Thank you.

**Comment by kyrios123** [comment]
Also I see libraries that are installed in `/usr/share/projectM/presets` ???

```
lrwxrwxrwx 1 build build     30 Aug 17 16:18  libMstressJuppyDancer.so -> libMstressJuppyDancer.so.0.0.0
lrwxrwxrwx 1 build build     30 Aug 17 16:18  libMstressJuppyDancer.so.0 -> libMstressJuppyDancer.so.0.0.0
-rwxr-xr-x 1 build build  18400 Aug 17 16:19  libMstressJuppyDancer.so.0.0.0
lrwxrwxrwx 1 build build     28 Aug 17 16:18  libRLGFractalDrop7c.so -> libRLGFractalDrop7c.so.0.0.0
lrwxrwxrwx 1 build build     28 Aug 17 16:18  libRLGFractalDrop7c.so.0 -> libRLGFractalDrop7c.so.0.0.0
-rwxr-xr-x 1 build build  18400 Aug 17 16:19  libRLGFractalDrop7c.so.0.0.0
lrwxrwxrwx 1 build build     30 Aug 17 16:18  libRovastarDarkSecret.so -> libRovastarDarkSecret.so.0.0.0
lrwxrwxrwx 1 build build     30 Aug 17 16:18  libRovastarDarkSecret.so.0 -> libRovastarDarkSecret.so.0.0.0
-rwxr-xr-x 1 build build  18400 Aug 17 16:19  libRovastarDarkSecret.so.0.0.0
lrwxrwxrwx 1 build build     33 Aug 17 16:18  libRovastarDriftingChaos.so -> libRovastarDriftingChaos.so.0.0.0
lrwxrwxrwx 1 build build     33 Aug 17 16:18  libRovastarDriftingChaos.so.0 -> libRovastarDriftingChaos.so.0.0.0
-rwxr-xr-x 1 build build  22496 Aug 17 16:19  libRovastarDriftingChaos.so.0.0.0
lrwxrwxrwx 1 build build     33 Aug 17 16:18  libRovastarFractalSpiral.so -> libRovastarFractalSpiral.so.0.0.0
lrwxrwxrwx 1 build build     33 Aug 17 16:18  libRovastarFractalSpiral.so.0 -> libRovastarFractalSpiral.so.0.0.0
-rwxr-xr-x 1 build build  14304 Aug 17 16:19  libRovastarFractalSpiral.so.0.0.0
lrwxrwxrwx 1 build build     36 Aug 17 16:18  libRovastarFractopiaFrantic.so -> libRovastarFractopiaFrantic.so.0.0.0
lrwxrwxrwx 1 build build     36 Aug 17 16:18  libRovastarFractopiaFrantic.so.0 -> libRovastarFractopiaFrantic.so.0.0.0
-rwxr-xr-x 1 build build  18400 Aug 17 16:19  libRovastarFractopiaFrantic.so.0.0.0
```

**Comment by kyrios123** [comment]
@revmischa before talking about PR, I need to succeed running projectM. :wink: 
I haven't had time to dig yet but here is what I get when I execute the binaries (`projectMSDL` and `projectM-pulseaudio`)

```
$ projectMSDL 
INFO: GL_VERSION: (null)
INFO: GL_SHADING_LANGUAGE_VERSION: (null)
INFO: GL_VENDOR: (null)
[projectM] config file: /usr/share/projectM/config.inp
Floating point exception
 

$ projectM-pulseaudio 

(projectM-pulseaudio:23680): Gtk-WARNING **: 16:24:53.384: Unable to locate theme engine in module_path: "murrine",

(projectM-pulseaudio:23680): Gtk-WARNING **: 16:24:53.385: Unable to locate theme engine in module_path: "murrine",

(projectM-pulseaudio:23680): Gtk-WARNING **: 16:24:53.385: Unable to locate theme engine in module_path: "murrine",
dir:/usr/share/projectM/config.inp 
reading ~/.projectM/config.inp 
[projectM] config file: /home/py/.projectM/config.inp
Error compiling base vertex shader: 0:1(10): error: GLSL 4.10 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES

Error compiling base fragment shader: 0:1(10): error: GLSL 4.10 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES

error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader
Error compiling base vertex shader: 0:1(10): error: GLSL 4.10 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES

Error compiling base fragment shader: 0:1(10): error: GLSL 4.10 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES

error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader
No Textures Loaded from /usr/local/share/projectM/textures
[projectM] Allocating idle preset...
Loading preset idle://Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] path is Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] url is idle://Geiss & Sperl - Feedback (projectM idle HDR mix).milk
unconnected: connecting...
connectHelper:  "alsa_output.pci-0000_00_05.0.analog-stereo.monitor"
CREATED
READY
[PresetFactory] path is r/share/projectM/presets/illusion & studio music - charged bliss.milk
[PresetFactory] url is /usr/share/projectM/presets/illusion & studio music - charged bliss.milk
```

**Comment by revmischa** [comment]
What system is this on? Try building with --enable-gles?

**Comment by revmischa** [comment]
It looks like you're building the pre-release beta, right?

**Comment by kyrios123** [comment]
Nope, I'm building the 3.0.0 final. It works indeed much better with `--enable-gles`
```
$ projectMSDL 
INFO: GL_VERSION: 3.1 Mesa 18.1.6
INFO: GL_SHADING_LANGUAGE_VERSION: 1.40
INFO: GL_VENDOR: VMware, Inc.
[projectM] config file: /usr/share/projectM/config.inp
No Textures Loaded from /usr/local/share/projectM/textures
[projectM] Allocating idle preset...
Loading preset idle://Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] path is Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] url is idle://Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] path is r/share/projectM/presets/Geiss - Heavenly 1.milk
[PresetFactory] url is /usr/share/projectM/presets/Geiss - Heavenly 1.milk
INFO: Using audio driver: pulseaudio
INFO: Found audio capture device 0: Built-in Audio Analog Stereo
INFO: Opened audio capture device 2: Built-in Audio Analog Stereo
INFO: Sample rate: 512, frequency: 48000, channels: 2, format: 33056
[PresetFactory] path is r/share/projectM/presets/Rovastar & Krash - Cerebral Demons (Beat Pulse Mix).milk
[PresetFactory] url is /usr/share/projectM/presets/Rovastar & Krash - Cerebral Demons (Beat Pulse Mix).milk
[PresetFactory] path is r/share/projectM/presets/Zylot - Rainbow Planet Under Attack.milk
[PresetFactory] url is /usr/share/projectM/presets/Zylot - Rainbow Planet Under Attack.milk
[PresetFactory] path is r/share/projectM/presets/Rovastar + Geiss - Hyperkaleidoscope Glow 2 motion blur.milk
[PresetFactory] url is /usr/share/projectM/presets/Rovastar + Geiss - Hyperkaleidoscope Glow 2 motion blur.milk
[PresetFactory] path is r/share/projectM/presets/Geiss - Reaction Diffusion 2.milk
[PresetFactory] url is /usr/share/projectM/presets/Geiss - Reaction Diffusion 2.milk
[PresetFactory] path is r/share/projectM/presets/Rovastar & Geiss - Dynamic Swirls 3 (Smoking Delusion Mix).milk
[PresetFactory] url is /usr/share/projectM/presets/Rovastar & Geiss - Dynamic Swirls 3 (Smoking Delusion Mix).milk
[PresetFactory] path is r/share/projectM/presets/Illusion & Rovastar - Clouded Bottle.milk
[PresetFactory] url is /usr/share/projectM/presets/Illusion & Rovastar - Clouded Bottle.milk
[PresetFactory] path is r/share/projectM/presets/Geiss - Feedback 2.milk
[PresetFactory] url is /usr/share/projectM/presets/Geiss - Feedback 2.milk
[PresetFactory] path is r/share/projectM/presets/Tschoey - Music Flower.milk
[PresetFactory] url is /usr/share/projectM/presets/Tschoey - Music Flower.milk
[PresetFactory] path is r/share/projectM/presets/Geiss - Bass Kaleidosphere.milk
[PresetFactory] url is /usr/share/projectM/presets/Geiss - Bass Kaleidosphere.milk
[PresetFactory] path is r/share/projectM/presets/Flexi, Rovastar + Geiss - Fractopia vs bas relief.milk
[PresetFactory] url is /usr/share/projectM/presets/Flexi, Rovastar + Geiss - Fractopia vs bas relief.milk
[PresetFactory] path is r/share/projectM/presets/Krash - molten indecision.milk
[PresetFactory] url is /usr/share/projectM/presets/Krash - molten indecision.milk
[projectM] thread cleanup
[~MilkdropPresetFactory] destroy infix ops
[~MilkdropPresetFactory] destroy builtin func
[~MilkdropPresetFactory] delete preset out puts
[~MilkdropPresetFactory] done

$ projectM-pulseaudio 

(projectM-pulseaudio:15597): Gtk-WARNING **: 16:50:22.564: Unable to locate theme engine in module_path: "murrine",

(projectM-pulseaudio:15597): Gtk-WARNING **: 16:50:22.565: Unable to locate theme engine in module_path: "murrine",

(projectM-pulseaudio:15597): Gtk-WARNING **: 16:50:22.567: Unable to locate theme engine in module_path: "murrine",
dir:/usr/share/projectM/config.inp 
reading ~/.projectM/config.inp 
[projectM] config file: /home/py/.projectM/config.inp
No Textures Loaded from /usr/local/share/projectM/textures
[projectM] Allocating idle preset...
Loading preset idle://Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] path is Geiss & Sperl - Feedback (projectM idle HDR mix).milk
[PresetFactory] url is idle://Geiss & Sperl - Feedback (projectM idle HDR mix).milk
unconnected: connecting...
connectHelper:  "alsa_input.pci-0000_00_05.0.analog-stereo"
CREATED
READY
[PresetFactory] path is r/share/projectM/presets/fiShbRaiN + geiss - witchcraft (Grow Mix 3).milk
[PresetFactory] url is /usr/share/projectM/presets/fiShbRaiN + geiss - witchcraft (Grow Mix 3).milk
[PresetFactory] path is r/share/projectM/presets/Geiss - Bipolar 2.milk
[PresetFactory] url is /usr/share/projectM/presets/Geiss - Bipolar 2.milk
```

**Comment by revmischa** [comment]
The issue may (or may not) be fixed in the `glsl` branch (aka the beta release). We've been working on GLES and GLSL support in it.

**Comment by revmischa** [comment]
Is this still an issue with latest master or release?

**Comment by kyrios123** [comment]
Issue is fixed with version 3.1.0.
I have made [a patch](https://dev.getsol.us/D4382) for adding projectM in the Solus repository.
Thanks again for the help & follow up, @revmischa ! :+1:
